app/src/main/java/org/wikipedia/gallery/GalleryActivity.kt

```diff
@@ -0,0 +1,761 @@
+import org.wikipedia.page.linkpreview.LinkPreviewDialog
+import org.wikipedia.suggestededits.PageSummaryForEdit
+import org.wikipedia.suggestededits.SuggestedEditsImageTagEditActivity
+import org.wikipedia.suggestededits.SuggestedEditsSnackbars
+import org.wikipedia.theme.Theme
+import org.wikipedia.util.*
+import org.wikipedia.util.log.L
+import org.wikipedia.views.PositionAwareFragmentStateAdapter
+import org.wikipedia.views.ViewAnimations
+import org.wikipedia.views.ViewUtil
+import java.io.File
+import java.util.*
+
+class GalleryActivity : BaseActivity(), LinkPreviewDialog.Callback, GalleryItemFragment.Callback {
+
+    private lateinit var binding: ActivityGalleryBinding
+    private lateinit var sourceWiki: WikiSite
+    private lateinit var funnel: GalleryFunnel
+    private var pageChangeListener: GalleryPageChangeListener? = GalleryPageChangeListener()
+    private var pageTitle: PageTitle? = null
+    private var imageEditType: ImageEditType? = null
+    private val disposables = CompositeDisposable()
+    private var imageCaptionDisposable: Disposable? = null
+    private var revision = 0L
+    private var controlsShowing = true
+    /**
+     * If we have an intent that tells us a specific image to jump to within the gallery,
+     * then this will be non-null.
+     */
+    private var initialFilename: String? = null
+    /**
+     * If we come back from savedInstanceState, then this will be the previous pager position.
+     */
+    private var initialImageIndex = -1
+    private var galleryAdapter: GalleryItemAdapter? = null
+    private var targetLanguageCode: String? = null
+    private val app = WikipediaApp.getInstance()
+    private val bottomSheetPresenter = ExclusiveBottomSheetPresenter()
+    private val downloadReceiver = MediaDownloadReceiver()
+    private val downloadReceiverCallback: MediaDownloadReceiverCallback = MediaDownloadReceiverCallback()
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityGalleryBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        setSupportActionBar(binding.toolbar)
+        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
+        supportActionBar!!.title = ""
+        setNavigationBarColor(Color.BLACK)
+        binding.toolbarGradient.background = GradientUtil.getPowerGradient(R.color.black26, Gravity.TOP)
+        binding.infoGradient.background = GradientUtil.getPowerGradient(R.color.black38, Gravity.BOTTOM)
+        binding.descriptionText.movementMethod = linkMovementMethod
+        binding.creditText.movementMethod = linkMovementMethod
+        (binding.errorView.findViewById<View>(R.id.view_wiki_error_icon) as ImageView)
+            .setColorFilter(ContextCompat.getColor(this, R.color.base70))
+        (binding.errorView.findViewById<View>(R.id.view_wiki_error_text) as TextView)
+            .setTextColor(ContextCompat.getColor(this, R.color.base70))
```

Comment:
Optional but it should be better if we create a method in `ErrorView` to update the colors.